### PR TITLE
ADCM-2648 Adopt tests for new RBAC view logic

### PR DESCRIPTION
**Pull request by Sealwing**
Main changes:
- `is_denied` logic is changed to check permissions on interactions with objects directly via API, not via the client (ADCM client is checked the way it was)
- "user" objects are now received not at the start of the test and only those objects that are requested are the ones the user has permission for

Pay attention to:
- `is_denied` requires either ADCM client as `adcm_object` or ADCM object + client (by kwarg) to perform the correct check
- `is_allowed` operates only on "non-superuser" user objects it the alternative isn't set directly

### CODE REVIEW

**Review by alexterent — APPROVED** [review]

File: tests/functional/rbac/checkers.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+)
+
+from tests.library.consts import HTTPMethod
+from tests.functional.tools import AnyADCMObject, AnyRBACObject
+
+RoleTargetObject = Union[AnyADCMObject, AnyRBACObject, Bundle, ADCM]
+RoleTargetType = Type[RoleTargetObject]
+
+
+class ForbiddenCallChecker:  # pylint: disable=too-few-public-methods
```
Comment (annademina) [inline tests/functional/rbac/checkers.py:46]: need docstrings to class and method to describe this

File: tests/functional/rbac/checkers.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        Cluster: 'cluster/{id}/',
+        Service: 'cluster/{cluster_id}/service/{id}/',
+        Component: 'cluster/{cluster_id}/service/{service_id}/component/{id}/',
+        Provider: 'provider/{id}/',
+        Host: 'provider/{provider_id}/host/{id}/',
+        User: 'rbac/user/{id}/',
+        Group: 'rbac/group/{id}/',
+        Role: 'rbac/role/{id}/',
+        Policy: 'rbac/policy/{id}/',
+        # TODO jobs, tasks?
```
[the remark this answers is not in the document]
Comment (Sealwing) [reply tests/functional/rbac/checkers.py:65]: I'll make it a regular comment just to mark the place where to add new entities, when they'll be needed

File: tests/functional/rbac/checkers.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    def __init__(
+        self,
+        object_type: Union[RoleTargetType, Collection[RoleTargetType]],
+        endpoint_suffix: str,
+        method: HTTPMethod,
+        *,
+        special_case: Optional[str] = None,
+    ):
+        if special_case:
+            # rework to switch maybe if you have time
```
[the remark this answers is not in the document]
Comment (Sealwing) [reply tests/functional/rbac/checkers.py:81]: I don't think this one needs a separate task, it's just an idea for someone who'll find this one difficult to use

File: tests/functional/rbac/checkers.py (hunk 1 above)
Comment (Sealwing) [reply tests/functional/rbac/checkers.py:46]: Added docstring to `__call__`, not sure where there's a need for more documentation

**Review by annademina — APPROVED** [review]

**Pull request merged**